### PR TITLE
Add access to GC runtime profile stats

### PR DESCRIPTION
**Pull request by ZombineDev**
A rebase of #2155

### CODE REVIEW

File: src/core/memory.d [hunk 1]
```diff
@@ -170,6 +171,23 @@ struct GC
         size_t freeSize;
     }
 
+    /**
+     * Aggregation of current profile information
+     */
+    static struct ProfileStats {
```
Comment (jacob-carlborg) [inline src/core/memory.d]: Curly braces on their own lines.

**Comment by thewilsonator** [comment]
@ZombineDev ping.

**Comment by ZombineDev** [comment]
@thewilsonator In the short term, I don't think I'll be able to dedicate the time this PR deserves to get through the finish line. Can you adopt it?

**Comment by thewilsonator** [comment]
rebased, we'll see what happens.

**Comment by rainers** [comment]
I've pushed an update to mak/SRC and mak/COPY for missing files. It seems the compiler adds a ModuleInfo reference to a module that is not on the command line.

**Comment by rainers** [comment]
Rebased, squashed and added a micro-test.

BTW: the original PR was already approved by Andrei: https://github.com/dlang/druntime/pull/2155#pullrequestreview-108505370

**Pull request merged**